sale_report_delivered/reports/sale_report.py

```diff
@@ -164,8 +164,15 @@ def _from(self):
         return from_str
 
     def _where(self):
+        """Take into account only stock moves from internal locations to other
+        locations and moves from customer with the field 'to_refund' True
+        """
         return """
-            WHERE sm.state = 'done' OR sm.state IS NULL
+            WHERE (sm.state = 'done' OR sm.state IS NULL) AND (
+                (source_location.usage = 'internal' AND dest_location.usage = 'customer') OR
+                (source_location.usage = 'customer' AND dest_location.usage = 'internal'
+                    AND sm.to_refund)
```

Comment:
This leaves out dropshipping, isn't it?

Reply:
You're right.
A new PR could be made by changing both `usage = 'internal'` to `usage IN ('internal', 'supplier')`